src/adapter/graphql/src/queries/datasets/dataset_flow_runs.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use futures::TryStreamExt;
+use {kamu_flow_system as fs, opendatafabric as odf};
+
+use crate::prelude::*;
+use crate::queries::Flow;
+use crate::utils;
+
+///////////////////////////////////////////////////////////////////////////////
+
+pub struct DatasetFlowRuns {
+    dataset_handle: odf::DatasetHandle,
+}
+
+#[Object]
+impl DatasetFlowRuns {
+    const DEFAULT_PER_PAGE: usize = 15;
+
+    #[graphql(skip)]
+    pub fn new(dataset_handle: odf::DatasetHandle) -> Self {
+        Self { dataset_handle }
+    }
+
+    pub async fn view_flows(
```

Comment:
would recommend naming `list_flows` for consistency e.g. with `listTasksByDataset`

Reply:
Let it be